plugins/flatpak/gs-flatpak.c

```diff
@@ -2648,19 +2708,13 @@ gs_flatpak_get_list_for_install (GsFlatpak *self, GsApp *app,
 	if (!gs_flatpak_refine_runtime_for_install (self, app, cancellable, error))
 		return FALSE;
 	runtime = gs_app_get_update_runtime (app);
-	if (runtime != NULL) {
-		g_autofree gchar *ref_display = NULL;
-		ref_display = gs_flatpak_app_get_ref_display (runtime);
-		if (g_hash_table_contains (hash_installed, ref_display)) {
-			g_debug ("%s is already installed, so skipping",
-				 gs_app_get_id (runtime));
-		} else {
-			g_debug ("%s/%s is not already installed, so installing",
-				 gs_flatpak_app_get_ref_name (runtime),
-				 gs_flatpak_app_get_ref_branch (runtime));
-			gs_app_list_add (list, runtime);
-		}
-	}
+	if (runtime != NULL)
+		gs_flatpak_add_app_to_list_if_not_installed (runtime, list, hash_installed);
+
+	/* add services flatpak */
+	services = gs_flatpak_get_services_app_for_runtime (self->plugin, runtime);
+	if (services != NULL)
+		gs_flatpak_add_app_to_list_if_not_installed (services, list, hash_installed);
```

Comment:
I understand the consistence with the code above and not repeating two blocks. However, this involves more changes to `gs-flatpak` than I really wanted. In this case, the code you're abstracting `gs_flatpak_add_app_to_list_if_not_installed` is only worth it because of the comments, but the comments in the function look out of place to me (they're referring to the installation). So just repeat the code, or maybe do use the function but return a boolean and use it to print the debug messages from outside.

Reply:
I'm not sure I understand what you're asking; I can't figure out which comments you're referring to.

Reply:
OK, I tried repeating the code but it looks wrong to me... I still get the feeling I am misinterpreting your comment and looking at the wrong thing, but if you don't feel strongly about it then we can just drop it.